test/remote-syslog.sh

```diff
@@ -415,3 +415,70 @@ fi
+
+    os::log::debug "$( oc set env daemonset/logging-fluentd USE_REMOTE_SYSLOG=true REMOTE_SYSLOG_HOST=$myhost REMOTE_SYSLOG_PORT=${ALTPORT} REMOTE_SYSLOG_USE_RECORD=true REMOTE_SYSLOG_SEVERITY=info REMOTE_SYSLOG_TAG_KEY- )"
+    os::log::debug "$( oc label node --all logging-infra-fluentd=true --overwrite=true )"
+    os::cmd::try_until_text "oc get pods -l component=fluentd" "^logging-fluentd-.* Running "
+
+    mypod=$( get_running_pod fluentd )
+else
+    # make sure fluentd is running after previous test
+    os::cmd::try_until_text "oc get pods -l component=fluentd" "^logging-fluentd-.* Running "
+    os::log::debug "$( oc label node --all logging-infra-fluentd- )"
+    os::cmd::try_until_text "oc get daemonset/logging-fluentd -o jsonpath='{ .status.numberReady }'" "0" $FLUENTD_WAIT_TIME
+
+    os::log::debug "$( oc set env daemonset/logging-fluentd FORWARD_INPUT_LOG_LEVEL=info )"
+    os::log::debug "$( oc label node --all logging-infra-fluentd=true --overwrite=true )"
+    os::cmd::try_until_text "oc get pods -l component=fluentd" "^logging-fluentd-.* Running "
+
+    # make sure mux is running after previous test
+    os::cmd::try_until_text "oc get pods -l component=mux" "^logging-mux.* Running "
+    os::log::debug "$( oc scale --replicas=0 dc logging-mux )"
+    os::cmd::try_until_text "oc get dc logging-mux -o jsonpath='{ .status.replicas }'" "0" $MUX_WAIT_TIME
+    os::log::debug "$( oc set env dc/logging-mux FORWARD_INPUT_LOG_LEVEL=info USE_REMOTE_SYSLOG=true REMOTE_SYSLOG_HOST=$myhost REMOTE_SYSLOG_PORT=${ALTPORT} REMOTE_SYSLOG_USE_RECORD=true REMOTE_SYSLOG_SEVERITY=info REMOTE_SYSLOG_TAG_KEY- )"
+    os::log::debug "$( oc scale --replicas=1 dc logging-mux )"
+    os::cmd::try_until_text "oc get pods -l component=mux" "^logging-mux-.* Running " $MUX_WAIT_TIME
+
+    mypod=$( get_running_pod mux )
+fi
+os::cmd::try_until_success "oc exec $mypod find /etc/fluent/configs.d/dynamic/output-remote-syslog.conf" $MUX_WAIT_TIME
+
+artifact_log $title $mypod
+
+mymessage="rsyslogTestMessage-"$( date +%Y%m%d-%H%M%S )
+logger -i -p local1.err -t rsyslogTestTag $mymessage
+es_pod=$( get_es_pod es )
+es_ops_pod=$( get_es_pod es-ops )
+es_ops_pod=${es_ops_pod:-$es_pod}
+qs='{"query":{"term":{"systemd.u.SYSLOG_IDENTIFIER":"rsyslogTestTag"}}}'
+if os::cmd::try_until_text "curl_es ${es_ops_pod} /.operations.*/_count -X POST -d '$qs' | get_count_from_json" 1 $MUX_WAIT_TIME; then
+    artifact_log good - found $mymessage
+else
+    artifact_log failed - not found $mymessage
+fi
+os::cmd::try_until_text "sudo egrep \"${mymessage}$\" /var/log/messages" ".*${mymessage}.*" $MUX_WAIT_TIME
+artifact_log Log test message by logger: $mymessage
+sudo egrep "${mymessage}$" /var/log/messages 2>&1 | artifact_out || :
+
+oc logs $mypod 2>&1 | artifact_out || :
+no_tag_key_log=$( mktemp )
+oc logs $mypod > $no_tag_key_log 2>&1 || :
+grep NoMethodError $no_tag_key_log > /dev/null || :
```

Comment:
calling anything then doing `|| :` will always set `$?` to 0:
```
>foo() { return 1 ; }
>foo
>echo $?
1
>foo || :
>echo $?
0
```
What should this test do?  Is it sufficient to look for the message in elasticsearch and in /var/log/mesages?